### ISSUE TITLE
Zowe CLI Daemon w/JSON Config No Longer Works

### ISSUE THREAD

**Issue by dkelosky** [zowe/zowe-cli#1232]
For this version:
```
C:\dev>zowe --version
7.0.0-next.202112021313
```

Steps to reproduce:

1. `zowe config init` and respond to credential prompts
2. fill in `host` and `port` for `zosmf` profile in `zowe.config.json`
3. `set ZOWE_USE_DAEMON=false`
4. sanity test `zowe jobs list jobs`
5. `set ZOWE_USE_DAEMON=true`
6. run `zowe jobs list jobs` again

Receive prompt:
```
C:\dev>zowe jobs list jobs
Enter the user name for your service (will be hidden):
```

I believe this used to work.  Is this a new bug?

**Comment by t1m0thyj** [comment]
Thanks for reporting! Confirmed this is a regression introduced in https://github.com/zowe/imperative/commit/85b905b59d36184fe9669471804d7493919c2eb4

**Comment by t1m0thyj** [comment]
This was fixed in `@zowe/imperative@5.0.0-next.202112101814` and `@zowe/cli@7.0.0-next.202112102143`.
